Tasks/Maven/CodeAnalysis/mavencodeanalysis.ts

```diff
@@ -120,17 +86,64 @@ function findCandidateModules(directory:string):ModuleAnalysis[] {
         }
     });
 
+    // Special case: if the root turns up as a module, the automatic name won't do
+    result.forEach((module:ModuleAnalysis) => {
+        if (module.rootDirectory == tl.getVariable('build.sourcesDirectory')) {
+            module.moduleName = 'root';
+        }
+    });
+
     return result;
 }
 
-// Discover analysis results from enabled tools and associate them with the modules they came from
-function processAndAssignAnalysisResults(enabledCodeAnalysisTools:string[], modules:ModuleAnalysis[]):ModuleAnalysis[] {
+export function getCodeAnalysisStagingDirectory() {
```

Comment:
We do not, at this time, have a common code analysis module. This is expected to be done as part of work on PMD for Gradle.

Renaming the function - accepted.